src/jvm/test/org/jetbrains/kotlinx/lincheck_test/strategy/modelchecking/snapshot/ArraysAPISnapshotTest.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2024 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.strategy.modelchecking.snapshot
+
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.execution.ExecutionResult
+import org.jetbrains.kotlinx.lincheck.execution.ExecutionScenario
+import org.jetbrains.kotlinx.lincheck.strategy.managed.ManagedOptions
+import org.junit.Ignore
+import java.util.Arrays
+import kotlin.random.Random
+
+private val arrayValue = intArrayOf(2, 1, 4, 3, 6, 5, 8, 7, 10, 9)
+
+// TODO: parallel operations are not supported because of java.lang.ClassCastException:
```

Comment:
Should this TODO be left?

Reply:
`Arrays` API has `parallel*` versions of methods, I added this todo to add them later, when I fix the issue with `System.arraycopy`.

I updated the `TODO` message a little bit.